src/get-started/java-binding.md

```diff
@@ -314,38 +314,47 @@ common path corresponding to the service name. Thus, the `/balance/:walletId`
 handler for balance requests in the "cryptocurrency" service will be available
 at `/api/cryptocurrency/balance/:walletId`.
 
-See [documentation][vertx.io] on the possibilities of `Vert.x` used as a web
+See [documentation][vertx-web-docs] on the possibilities of `Vert.x` used as a web
 framework.
 
 ### Dependencies Management
 
 Exonum uses [Guice](https://github.com/google/guice) to describe the
 dependencies of the service components (both system-specific
-ones, for example, Exonum time service, and external ones). Each
-service should define a module describing implementations of the framework
-interfaces – `Service`, `TransactionConverter` and implementations of other
-components, if any.
+ones, for example, Exonum time service, and external ones).
+Each service should define a Guice module describing implementations of
+the `Service` and its dependencies, if any.
+
+A service module shall:
+
+  1. extend [`AbstractServiceModule`][abstract-service-module-javadoc]
+  2. be annotated with `@org.pf4j.Extension`.
+  3. be `public`.
 
 !!! note "Minimalistic Example of Service Module"
     ```java
-    public class ServiceModule extends AbstractModule {
+    @Extension
+    public class ServiceModule extends AbstractServiceModule {
 
       @Override
       protected void configure() {
         // Define the Service implementation.
         bind(Service.class).to(CryptocurrencyService.class).in(Singleton.class);
 
-        // Define the TransactionConverter implementation.
+        // Define the TransactionConverter implementation required
+        // by the CryptocurrencyService.
         bind(TransactionConverter.class).to(CryptocurrencyTransactionConverter.class);
       }
     }
     ```
 
-The fully-qualified name (FQN) of the module class must be specified/passed
-during configuration of an Exonum App that will run the service.
+The fully-qualified name of the module class is recorded in the service artifact
+metadata and is used by the framework to instantiate services.
 
 For more information on using Guice, see the [project wiki][Guice].
 
+[abstract-service-module-javadoc]: https://exonum.com/doc/api/java-binding-core/0.5.0/com/exonum/binding/service/AbstractServiceModule.html
```

Comment:
why the link is here, not in the end of file?

Reply:
To keep them local.